spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -252,7 +104,7 @@ public synchronized Object invoke(MethodInvocation invocation) throws Throwable
 	@Override
 	public synchronized Object getObject() throws Exception {
 		if (this.proxy == null) {
-			createChannels(this.type);
+			this.channelFactory.createChannels(this.type);
```

Comment:
This can be done at the end of `afterPropertiesSet()`, I believe - and it'll take care of the issue. What we want is for the channels to be already created when they're injected into their target beans.